harvester_e2e_tests/integration/test_rancher_integration.py

```diff
@@ -0,0 +1,319 @@
+                            "disableEviction": False,
+                            "enabled": False,
+                            "force": False,
+                            "gracePeriod": -1,
+                            "ignoreDaemonSets": True,
+                            "ignoreErrors": False,
+                            "skipWaitForDeleteTimeoutSeconds": 0,
+                            "timeout": 120
+                        }
+                    },
+                    "machineGlobalConfig": {
+                        "cni": "calico",
+                        "disable-kube-proxy": False,
+                        "etcd-expose-metrics": False,
+                        "profile": None
+                    },
+                    "machineSelectorConfig": [
+                        {
+                            "config": {
+                                "cloud-provider-config": f"secret://{FLEET_DEFAULT_NAMESPACE}:{unique_name}",  # noqa
+                                "cloud-provider-name": "harvester",
+                                "protect-kernel-defaults": False
+                            }
+                        }
+                    ],
+                    "etcd": {
+                        "disableSnapshots": False,
+                        "s3": None,
+                        "snapshotRetention": 5,
+                        "snapshotScheduleCron": "0 */5 * * *"
+                    },
+                    "registries": {
+                        "configs": {},
+                        "mirrors": {}
+                    },
+                    "machinePools": [
+                        {
+                            "name": "pool1",
+                            "etcdRole": True,
+                            "controlPlaneRole": True,
+                            "workerRole": True,
+                            "hostnamePrefix": f"{unique_name}-",
+                            "labels": {},
+                            "quantity": 1,
+                            "unhealthyNodeTimeout": "0m",
+                            "machineConfigRef": {
+                                    "kind": "HarvesterConfig",
+                                    "name": unique_name
+                            }
+                        }
+                    ]
+                },
+                "machineSelectorConfig": [
+                    {
+                        "config": {}
+                    }
+                ],
+                "kubernetesVersion": k8s_version,
+                "defaultPodSecurityPolicyTemplateName": "",
+                "cloudCredentialSecretName": cloud_credential['id'],
+                "localClusterAuthEndpoint": {
+                    "enabled": False,
+                    "caCerts": "",
+                    "fqdn": ""
+                }
+            }
+        )
+
+        assert 201 == code, (
+            f"Failed to create RKE2 MgmtCluster {unique_name} with error: {code}, {data}"
+        )
```

Comment:
need to add check points to make sure VMs are created, provisioning, and states.